app/src/main/kotlin/com/wire/android/ui/home/conversations/model/MessageTypes.kt

```diff
@@ -178,16 +178,21 @@ private fun mapToDisplayMentions(uiText: UIText, resources: Resources): Pair<Lis
     return if (uiText is UIText.DynamicString) {
         val stringBuilder: StringBuilder = StringBuilder(uiText.value)
         val mentions = uiText.mentions.sortedBy { it.start }.reversed()
-        val mentionList = mentions.map {
-            val mentionName = uiText.value.substring(it.start, it.start + it.length)
-            stringBuilder.insert(it.start + it.length, MENTION_MARK)
-            stringBuilder.insert(it.start, MENTION_MARK)
-            DisplayMention(
-                it.userId,
-                it.length,
-                it.isSelfMention,
-                mentionName
-            )
+        val mentionList = mentions.mapNotNull {
+            // secured crash for mentions caused by web when text without mentions contains mention data
+            if (it.start + it.length < uiText.value.length) {
```

Comment:
Wouldn't be ok to also check if the first character of the mention is `@` in text and maybe also if after the mention there's is a white space 🤔